lib/spack/spack/ci.py

```diff
@@ -546,6 +486,245 @@ def get_spec_filter_list(env, affected_pkgs):
+        """Generate the IR from the Spack CI configurations."""
+
+        jobs = self.ir["jobs"]
+
+        # Implicit job defaults
+        defaults = [
+            {
+                "build-job": {
+                    "script": [
+                        "cd {env_dir}",
+                        "spack env activate --without-view .",
+                        "spack ci rebuild",
+                    ],
+                }
+            },
+            {
+                "noop-job": {
+                    "script": [
+                        'echo "All specs already up to date, nothing to rebuild."',
+                    ],
+                }
+            },
+        ]
+
+        # Job overrides
+        overrides = [
+            # Reindex script
+            {
+                "reindex-job": {
+                    "script:": [
+                        "spack buildcache update-index --keys --mirror-url {index_target_mirror}",
+                    ],
+                }
+            },
+            # Cleanup script
+            {
+                "cleanup-job": {
+                    "script:": [
+                        "spack -d mirror destroy --mirror-url {mirror_prefix}/$CI_PIPELINE_ID",
+                    ]
+                }
+            },
+            # Add signing job tags
+            {
+                "signing-job": {
+                    "tags": ["aws", "protected", "notary"],
+                }
+            },
+            # Remove reserved tags
+            {
+                "any-job-remove": {
+                    "tags": SPACK_RESERVED_TAGS,
+                }
+            },
+        ]
+
+        pipeline_gen = overrides + self.ci_config.get("pipeline-gen", []) + defaults
+
+        for section in reversed(pipeline_gen):
+            name = self.__is_named(section)
+            has_submapping = "submapping" in section
+            section = cfg.InternalConfigScope._process_dict_keyname_overrides(section)
+
+            if name:
+                remove_job_name = self.__job_name(name, suffix="-remove")
+                merge_job_name = self.__job_name(name)
+                do_remove = remove_job_name in section
+                do_merge = merge_job_name in section
+
+                def _apply_section(dest, src):
+                    if do_remove:
+                        dest = spack.config.remove_yaml(dest, src[remove_job_name])
+                    if do_merge:
+                        dest = copy.copy(spack.config.merge_yaml(dest, src[merge_job_name]))
+
+                if name == "build":
+                    # Apply attributes to all build jobs
+                    for _, job in jobs.items():
+                        if job["spec"]:
+                            _apply_section(job["attributes"], section)
+                elif name == "any":
+                    # Apply section attributes too all jobs
+                    for _, job in jobs.items():
+                        _apply_section(job["attributes"], section)
```

Comment:
I was expecting a case for `noop-job` here, what am I missing?

Reply:
noop is handled via the fall through else